src/main/kotlin/com/cognifide/gradle/aem/environment/hosts/HostsOptions.kt

```diff
@@ -0,0 +1,33 @@
+package com.cognifide.gradle.aem.environment.hosts
+
+import com.cognifide.gradle.aem.environment.EnvironmentException
+import org.apache.commons.lang3.SystemUtils
+
+class HostsOptions {
+    var list = listOf(
+        Host("127.0.0.1", "example.com"),
+        Host("127.0.0.1", "demo.example.com"),
+        Host("127.0.0.1", "author.example.com"),
+        Host("127.0.0.1", "invalidation-only")
+    )
+
+    val file = if (SystemUtils.IS_OS_WINDOWS) {
+        HOSTS_FILE_ON_WINDOWS
+    } else {
+        HOSTS_FILE
+    }
+
+    fun configure(config: Map<String, String>) {
+        config.forEach { hostEntry ->
+            if (hostEntry.key.isBlank() || hostEntry.value.isBlank()) {
+                throw EnvironmentException("Invalid hosts configuration (empty value), ip: ${hostEntry.key}, name: ${hostEntry.value}. ")
```

Comment:
move that logic into Host init / constructor (more close to domain)
validate in construction / inside